kuma/javascript/src/article.jsx

```diff
@@ -114,7 +114,9 @@ export default function Article({ document }: DocumentProps) {
     useEffect(() => {
         let rootElement = article.current;
         if (rootElement) {
-            activateBCDSignals(document.slug, document.locale, userData);
+            if (userData) {
```

Comment:
I'm confused. The use of this depends on being logged in but it's not enforced on the server.

Reply:
@peterbe, The userData object is created even if the user is not logged in. I tested this on a [browser & computer](https://www.screencast.com/t/UNSO6kobXY) where I didn't ever sign in before. 

As far as I remember, the reason why I added this control, was that the activateBCDSignals component was rendered twice due to userData changing. The component gets rendered with null, then again with the object having a new structure and value (see [screenshot](https://www.screencast.com/t/uQrUh6o9mR4)). Then, if the userData object is not passed, the waffle flag inside bcd.js is not even evaluated (see [screenshot](https://www.screencast.com/t/uQrUh6o9mR4)), so the "What are we missing?" button doesn't show.

Reply:
So it's not really the user you're waiting for, it's the waffle flags. :)

I raised the topic of waffle elsewhere and I think we should redirect our attention to that discussion and when that's settled come back here. 

Also, the implementation should be fixed anyway because this effect is triggered by `[document, userData]` which makes me nervous. There might be something else that changes `document` or `userData` when in fact the only things that matter are `[document.slug, document.locale, userData.waffle.flags['bc-signals']]`.

To start with, remove any mention of "waffle" in `bcd.js` and make `activateBCDSignal` mean what it's called. E.g.
```javascript
bcTable = document && document.querySelector('.bc-table');
if (bcTable) {
    if (SOME WAY OF TELLING IF IT'S ALREADY *not* ACTIVATED) {
      const waffleEnabled = userData && userData.waffle.flags['bc-signals'];
      if (waffleEnabled) {
         activateBCDSignals(document.slug, document.locale)
      }
    }
}
```

Reply:
We can discuss the use of waffle elsewhere but the invokation needs to become idempotent so that it doesn't hurt to trigger this effect without accidentally activating it twice.